src/mlpack/methods/ann/layer/max_pooling.hpp

```diff
@@ -171,18 +171,34 @@ class MaxPoolingType : public Layer<MatType>
         for (size_t i = 0, rowidx = 0; i < output.n_rows;
             ++i, rowidx += strideWidth)
         {
+          size_t rowEnd = rowidx + kernelWidth - 1;
+          size_t colEnd = colidx + kernelHeight - 1;
+
+          if (rowEnd > input.n_rows - 1)
+          {
+            if (floor)
+              continue;
+            rowEnd = input.n_rows - 1;
+          }
```

Comment:
It took me a little while to actually understand how `floor` is being used but I get it now and agree that these changes are correct.  Nice!